### PR TITLE
Mobile: Implementation of Login Screen

### LINKED ISSUE #396: Mobile - Implementing Login Page
### Issue Description

I will be responsible for the login screen implementation. When this issue is done, login screen will completely functional. Unit test for the login page will done afterwards.

### Step Details

Steps that will be performed:
- [x]  Initialize login screen
- [x]  Add necessary widgets
- [x]  Add logo and title
- [x]  Set up the proper page navigations

### Final Actions

After the work is done, i will test the login screen via emulator. If there is no erroneous components and functionalities, results will be provided here and related PR will be opened.

### Deadline of the Issue

26.10.2022 - 23.59

### Reviewer

Ezgi Aysel Batı

### Deadline for the Review

27.10.2022 - 23.59

### PR DESCRIPTION
**Pull request by onurkomurcu**
I was responsible for login screen. Login screen is coherent with the sign up page to have consistent flow. It was tested via emulator. Test of the login page will be implemented in another PR. I will also responsible for that. Further details of the implementation should be reviewed before merging it.

Closes #396

### CODE REVIEW

**Review by bahricanyesil — CHANGES_REQUESTED** [review]

**Comment by bahricanyesil** [comment]
There is also an error related to the action button key but it appears on all screens including the signup. So, let's fix it with another PR.

**Review by bahricanyesil — CHANGES_REQUESTED** [review]
Please remove all GlobalKey<ActionButtonState> stuffs since we don't use them but it creates a problem for us.

**Review by ezgy — CHANGES_REQUESTED** [review]

**Comment by onurkomurcu** [comment]
I have considered whole feedbacks and made changes accordingly. It is ready to be reviewed.

File: learnify/mobile/lib/features/auth/login/view-model/login_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  Future<String?> _loginRequest() async {
+    final bool isValid = formKey.currentState?.validate() ?? false;
+    if (isValid) {
+      final LoginRequest requestModel = LoginRequest(
+          email: _emailController.text, password: _passwordController.text);
+      final IResponseModel<LoginResponse> res =
+          await _authService.login(requestModel);
+      if (res.hasError) return res.error?.errorMessage;
+      await navigationManager.navigateToPageClear(
+          path: NavigationConstants.home);
```
Comment (ezgy) [inline learnify/mobile/lib/features/auth/login/view-model/login_view_model.dart:78]: You shouldn't navigate to home when unverified users try to login. Instead, redirect them to verify screen. You can do that here once you get the response you can add a check for `if(res.data?.user?.isVerified == false)`

**Review by ezgy — CHANGES_REQUESTED** [review]

File: learnify/mobile/lib/features/auth/login/view-model/login_view_model.dart (hunk 1 above)
Comment (onurkomurcu) [reply learnify/mobile/lib/features/auth/login/view-model/login_view_model.dart]: That was an important case. I have changed accordingly. Thank you, @ezgy !

**Comment by onurkomurcu** [comment]
I'm waiting for a clarification of the response message of "not verified user login" case. If returns error message, i will change the code accordingly.

**Review by ezgy — APPROVED** [review]

**Review by egemenatikk — APPROVED** [review]
Looks good enough, approving.

**Pull request merged**